base/sort.jl

```diff
@@ -1240,7 +1535,7 @@ fpsort!(v::AbstractVector, a::Sort.PartialQuickSort, o::Ordering) =
 
 sort!(v::FPSortable, a::Algorithm, o::DirectOrdering) =
     fpsort!(v, a, o)
-sort!(v::AbstractVector{<:Integer}, a::Algorithm, o::Perm{<:DirectOrdering,<:FPSortable}) =
+sort!(v::AbstractVector{<:Union{Signed, Unsigned}}, a::Algorithm, o::Perm{<:DirectOrdering,<:FPSortable}) =
```

Comment:
Why change this? Even if `Bool` has a faster method, no need to disallow it as the most specific method will take precedence.

Reply:
Because the method actually doesn't work for Bool. This errors on master: `sort!(BitVector(undef, 10), order=Base.Perm(Base.Forward, [4.5, 2.0]))`.

Reply:
`ERROR: ArgumentError: invalid index: false of type Bool`